frame/support/src/dispatch.rs

```diff
@@ -2940,3 +3489,187 @@ mod tests {
 		Call::<TraitImpl>::new_call_variant_aux_0();
 	}
 }
+
+#[cfg(test)]
+// Do not complain about unused `dispatch` and `dispatch_aux`.
+#[allow(dead_code)]
+mod weight_tests {
+	use super::*;
+	use sp_core::{parameter_types, Get};
+	use sp_weights::RuntimeDbWeight;
+
+	pub trait Config: 'static {
+		type Origin;
+		type Balance;
+		type BlockNumber;
+		type DbWeight: Get<RuntimeDbWeight>;
+		type PalletInfo: crate::traits::PalletInfo;
+	}
+
+	pub struct TraitImpl {}
+
+	parameter_types! {
+		pub const DbWeight: RuntimeDbWeight = RuntimeDbWeight {
+			read: 100,
+			write: 1000,
+		};
+	}
+
+	impl Config for TraitImpl {
+		type Origin = u32;
+		type BlockNumber = u32;
+		type Balance = u32;
+		type DbWeight = DbWeight;
+		type PalletInfo = crate::tests::PanicPalletInfo;
+	}
+
+	decl_module! {
```

Comment:
should probably move this already to the new types, because someday we will deprecate `decl_module!`

Reply:
I think I should at least put a `#[deprecated]` on top of this.

Reply:
Actually no, bad idea, because some test code in our codebase still uses the `decl_module` macro, and we'd need to either `#[allow(deprecated)]` on those or remove those tests altogether.

Reply:
Should be a new tracking issue then to start actually deprecating `decl_module!`?

Reply:
https://github.com/paritytech/substrate/issues/12248